picard/ui/options/plugins.py

```diff
@@ -100,103 +127,182 @@ def __init__(self, parent=None):
-        self.ui.plugins.dropEvent = self.dropEvent
-        self.ui.plugins.dragEnterEvent = self.dragEnterEvent
-        if sys.platform == "win32":
-            self.loader = "file:///%s"
-        else:
-            self.loader = "file://%s"
+        plugins = self.ui.plugins
+
+        # fix for PICARD-1226, QT bug (https://bugreports.qt.io/browse/QTBUG-22572) workaround
+        plugins.setStyleSheet('')
+
+        plugins.itemSelectionChanged.connect(self.change_details)
+        plugins.mimeTypes = self.mimeTypes
+        plugins.dropEvent = self.dropEvent
+        plugins.dragEnterEvent = self.dragEnterEvent
+
         self.ui.install_plugin.clicked.connect(self.open_plugins)
         self.ui.folder_open.clicked.connect(self.open_plugin_dir)
         self.ui.reload_list_of_plugins.clicked.connect(self.reload_list_of_plugins)
-        self.tagger.pluginmanager.plugin_installed.connect(self.plugin_installed)
-        self.tagger.pluginmanager.plugin_updated.connect(self.plugin_updated)
-        self.ui.plugins.header().setStretchLastSection(False)
-        self.ui.plugins.header().setSectionResizeMode(COLUMN_NAME, QtWidgets.QHeaderView.Stretch)
-        self.ui.plugins.header().setSectionResizeMode(COLUMN_VERSION, QtWidgets.QHeaderView.Stretch)
-        self.ui.plugins.header().resizeSection(COLUMN_ACTION, 100)
-        self.ui.plugins.setSortingEnabled(True)
+
+        self.manager = self.tagger.pluginmanager
+        self.manager.plugin_installed.connect(self.plugin_installed)
+        self.manager.plugin_updated.connect(self.plugin_updated)
+
+        header = self.ui.plugins.header()
+        header.setStretchLastSection(False)
+        header.setSectionResizeMode(COLUMN_NAME, QtWidgets.QHeaderView.Stretch)
+        header.setSectionResizeMode(COLUMN_VERSION, QtWidgets.QHeaderView.Stretch)
+        header.resizeSection(COLUMN_ACTION, 100)
+        plugins.setSortingEnabled(True)
+
+    def items(self):
+        iterator = QTreeWidgetItemIterator(self.ui.plugins, QTreeWidgetItemIterator.All)
+        while iterator.value():
+            item = iterator.value()
+            iterator += 1
+            yield item
+
+    def find_item_by_plugin_name(self, plugin_name):
+        for item in self.items():
+            if plugin_name == item.plugin.module_name:
+                return item
+        return None
+
+    def selected_item(self):
+        try:
+            return self.ui.plugins.selectedItems()[0]
+        except IndexError:
+            return None
+
+    def selected_plugin(self):
+        item = self.selected_item()
+        if item:
+            return item.plugin
+        else:
+            return None
 
     def save_state(self):
         header = self.ui.plugins.header()
         config.persist["plugins_list_state"] = header.saveState()
         config.persist["plugins_list_sort_section"] = header.sortIndicatorSection()
         config.persist["plugins_list_sort_order"] = header.sortIndicatorOrder()
-        try:
```

Comment:
If I see this correctly the saving and restoring of `plugins_list_selected` selected did not actually work, and in your branch it is fixed. But do we actually need it? While using I found it rather confusing that I was in the middle of the list when opening the options dialog again.

Reply:
>  But do we actually need it? While using I found it rather confusing that I was in the middle of the list when opening the options dialog again.

Me too, i think i'll get rid of it, i think it was meant for the Reload List option, but it will be less disturbing if the list was at top each time

Reply:
Did not think about plugin reload, for that it actually could be useful. But still that should not require any real persisting, just temporarily remembering the position.

Reply:
> Did not think about plugin reload, for that it actually could be useful. But still that should not require any real persisting, just temporarily remembering the position.

That's what i did, check last iteration.